### PR TITLE
fix(wasm): Fix TextBox default alignment

### LINKED ISSUE #16568: [WASM] TextBox Vertical Content Alignment/Padding not behaving as expected
### Current behavior

The Fluent Textbox "vertical alignment" for its text differs between platforms:

WASM:
image

DESKTOP:
image

We can observe that as long as we change the FontSize or the Height attributes, the result will be different.
Probably it is the alignment in skia that is broken, since WASM version seems correct.

### Expected behavior

The alignment should be the same in all platforms

### How to reproduce it (as minimally and precisely as possible)

Using:
Uno.Templates::5.2.31

Run:
dotnet new unoapp -o TextBoxAlignment -preset "recommended" -platforms "android" -platforms "maccatalyst" -platforms "wasm" -platforms "windows" -platforms "desktop" -platforms "ios" -theme-service False -vscode False -pwa False -theme "fluent" -presentation "mvux" -config False -di False -log "none" -nav "blank" -http False -loc False -tests "none" -toolkit False -dsp False -id "com.companyname.TextBoxAlignment" -pub "O=TextBoxAlignment"

Add this code to MainPage.xaml:

```
<TextBox VerticalContentAlignment="Center" Text="Testing Vertical Text Alignment" />

<TextBox
	VerticalContentAlignment="Center"
	FontSize="11"
	Text="Testing FontSize 11" />


<TextBox
	Height="50"
	VerticalContentAlignment="Center"
	Text="Testing height 50" />

```

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #16568

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs [hunk 1]
```diff
@@ -49,7 +49,9 @@ private void OnForegroundChanged(DependencyPropertyChangedEventArgs e)
 		}
 
 		public TextBoxView(TextBox textBox, bool isMultiline)
-			: base(isMultiline ? "textarea" : "input")
+			// We need to use textarea regardless of isMultiline
+			// because "input" native HTML element can't have its text top-aligned.
+			: base("textarea")
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs:54]: This somehow breaks #16982 😕 

@MartinZikmund Do you see why?

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs]: That's an existing bug with master. If this XAML is used and long text is entered, then scrolling doesn't work.

```xml
<TextBox AcceptsReturn="true" Width="50" />
```

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs]: This is natively reproducible with:

```html
<!DOCTYPE html>
<html>
<body>

<textarea style="width: 50px; white-space: pre; overflow:hidden;">This is some very long text to test textarea native scrolling behavior.</textarea>
</body>
</html>

```

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs [hunk 2]
```diff
@@ -87,6 +91,11 @@ private protected override void OnLoaded()
 			HtmlInput += OnInput;
 			HtmlPaste += OnPaste;
 
+			if (!IsMultiline)
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.wasm.cs:94]: Or if not possible, a manual test.

**Comment by jeromelaban** [comment]
@Darsh0307 could you take a look at all the samples app tests related to textbox and testing the PR's test app compared to the latest master?

**Comment by Darsh0307** [comment]
I beleive so that 

The issue is related  to the textblock dosnet seems to be solved yet. 

Screenshot 2024-07-26 at 12 00 33 PM

**Comment by Youssef1313** [comment]
@Darsh0307 Could you clarify please?

**Comment by Darsh0307** [comment]
Hey @Youssef1313 I am just comparing the textblock>Input_TextScroll. 

I am comparing the Input_TextScroll between both Master and https://github.com/unoplatform/uno/pull/17665#issuecomment-2248266397

In that i compared both of them with XOR in paint.net 

By this we can clearly see that the allignment is still looking same.

**Comment by Youssef1313** [comment]
@Darsh0307 The alignment is not the same. It's now one pixel up, which is expected now that it's top-aligned instead of center

**Comment by Youssef1313** [comment]
image

Zoomed comparison:

image

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**